### PR TITLE
Personalized Caffeine Intake AI

### PR DESCRIPTION
**Pull request by SkaillZ**
Like any self-respecting SAAS application in the year of our lord 2024, integration with personalized AI™️ is necessary to maximize market disruption.

This feature requires `OPENAI_API_KEY` to be set in `credentials.php`. Otherwise, this feature is disabled.

Screenshot 2024-12-04 at 20 42 32

### CODE REVIEW

File: ai-summary.php [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+
+if (!isset($OPENAI_API_KEY)) {
+    throw new Error('Missing $OPENAI_API_KEY!');
+}
+
+// Fetch user-specific data
+$username = $_SESSION['username'];
+
+// Fetch user ID if logged in
+$userId = null;
```
Comment (meik99) [inline ai-summary.php:20]: @StreiblV should we maybe cache the userId in $_SESSION after a successful login so we don't have to repeat statements like that throughout the code base?

Comment (StreiblV) [reply ai-summary.php:20]: Yeah will happen during next cleanup session, more optimizations like this will happen soon.

**Comment by StreiblV** [comment]
@SkaillZ 
As far as I got that right, I have to pay as soon as I use this API, am I right?

**Comment by SkaillZ** [comment]
> @SkaillZ As far as I got that right, I have to pay as soon as I use this API?

Yes, testing this feature ~10 times has cost me $0.0018. You could alternatively switch the model to `gpt-3.5-turbo` and use a worse model for free.

**Pull request merged**